filter/store_test.go

```diff
@@ -0,0 +1,187 @@
+package filter_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/ONSdigital/dp-dataset-exporter/filter"
+	"github.com/ONSdigital/dp-dataset-exporter/filter/filtertest"
+	"github.com/ONSdigital/dp-dataset-exporter/observation"
+	. "github.com/smartystreets/goconvey/convey"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+)
+
+const filterAPIURL string = "http://filter-api:8765"
+const filterJobID string = "123456784432"
+
+var mockFilterData = &observation.Filter{
+	JobID:            filterJobID,
+	DataSetFilterID:  "123",
+	DimensionListURL: filterAPIURL + "/filter/" + filterJobID + "/dimensions",
+}
+
+var mockDimensionListData = []*observation.DimensionFilter{{
+	URL:  filterAPIURL + "/filter/" + filterJobID + "/dimensions/234",
+	Name: "Sex",
+}}
+
+var mockDimensionData = &observation.DimensionFilter{
+	URL:    filterAPIURL + "/filter/" + filterJobID + "/dimensions/234",
+	Name:   "Sex",
+	Values: []string{"Male", "Female"},
+}
+
+func TestStore_GetFilter(t *testing.T) {
+
+	mockFilterJSON, _ := json.Marshal(mockFilterData)
+	mockFilterBody := iOReadCloser{bytes.NewReader(mockFilterJSON)}
+
+	mockDimensionListJSON, _ := json.Marshal(mockDimensionListData)
+	mockDimensionListBody := iOReadCloser{bytes.NewReader(mockDimensionListJSON)}
+
+	mockDimensionJSON, _ := json.Marshal(mockDimensionData)
```

Comment:
I usually ignore them for the sake of keeping tests minimal. As the marshal is part of the test setup I figure that this would only become an issue when changing the tests, rather than an application change causing a failure.